### PR TITLE
change constraints setting value

### PR DESCRIPTION
**Pull request by xiewenlong1202**
getDisplayMedia did not support exact value

### CODE REVIEW

File: src/sdk/base/mediastream-factory.js [hunk 1]
```diff
@@ -181,7 +181,11 @@ export class MediaStreamFactory {
         };
       }
     } else {
-      mediaConstraints.audio = constraints.audio;
+      if (constraints.audio.source === MediaFormatModule.AudioSourceInfo.SCREENCAST) {
+        mediaConstraints.audio = true;
```
Comment (jianjunz) [inline src/sdk/base/mediastream-factory.js]: Please respect user's input. Why it cannot be false?

File: src/sdk/base/mediastream-factory.js [hunk 2]
```diff
@@ -198,11 +202,9 @@ export class MediaStreamFactory {
       if (constraints.video.resolution &&
           constraints.video.resolution.width &&
           constraints.video.resolution.height) {
-        mediaConstraints.video.width = Object.create({});
-        mediaConstraints.video.width.exact =
+        mediaConstraints.video.width =
```
Comment (jianjunz) [inline src/sdk/base/mediastream-factory.js]: It changes SDK's behavior. `exact` is what we expected.

Comment (xiewenlong1202) [reply src/sdk/base/mediastream-factory.js]: But when use the `exact` value, error will happen
```
TypeError: Failed to execute 'getDisplayMedia' on 'MediaDevices': exact constraints are not supported
```

File: src/sdk/base/mediastream-factory.js (hunk 1 above)
Comment (jianjunz) [reply src/sdk/base/mediastream-factory.js]: `constraints.audio.source === MediaFormatModule.AudioSourceInfo.SCREENCAST` means user is requesting audio. Resolve this one.

Comment (xiewenlong1202) [reply src/sdk/base/mediastream-factory.js]: I think when the audio source set to `screen-cast`, that is equal to `true`

File: src/sdk/base/mediastream-factory.js (hunk 2 above)
Comment (xiewenlong1202) [reply src/sdk/base/mediastream-factory.js]: Change to use 'number' only when share screen is needed.

**Pull request merged**